spec/models/user_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Omar') }
+  before { subject.save }
+
+  it 'Name should be present' do
+    subject.name = nil
+    expect(subject).to_not(be_valid)
+  end
+
+  it "shouldn't have any recent posts" do
+    expect(subject.recent_posts.length).to eq(0)
+  end
+
+  it 'should have a positive posts counter' do
+    subject.posts_counter = -1
+    expect(subject).to_not(be_valid)
+  end
+
+  it 'should have a positive posts counter' do
+    subject.posts_counter = -1
+    expect(subject).to_not(be_valid)
+  end
+
+  it 'displays three recent posts when called' do
+    4.times.map do |i|
+      post = subject.posts.create(title: "Post #{i}")
+      post.save
+      post
+    end
+  end
```

Comment:
- [ ] I fail to see how this block verifies that the recent posts method is tested. The method is not called and neither is there an assertion that the recent posts are fetched (you may need to create some posts). Kindly go fix your test and ensure the recent posts method is tested.